indexer/services/roundtable/src/tasks/update-affiliate-info.ts

```diff
@@ -0,0 +1,50 @@
+import { logger, stats } from '@dydxprotocol-indexer/base';
+import {
+  PersistentCacheTable, AffiliateInfoTable, PersistentCacheKeys, PersistentCacheFromDatabase,
+} from '@dydxprotocol-indexer/postgres';
+import { DateTime } from 'luxon';
+
+import config from '../config';
+
+const defaultLastUpdateTime: string = '2023-10-26T00:00:00Z';
+
+/**
+ * Update the affiliate info for all affiliate addresses.
+ */
+export default async function runTask(): Promise<void> {
+  try {
+    const start = Date.now();
+    const persistentCacheEntry: PersistentCacheFromDatabase | undefined = await PersistentCacheTable
+      .findById(PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME);
+
+    if (!persistentCacheEntry) {
+      logger.info({
+        at: 'update-affiliate-info#runTask',
+        message: `No previous ${PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME} found in persistent cache table. Will use default value: ${defaultLastUpdateTime}`,
+      });
+    }
+
+    const lastUpdateTime: DateTime = DateTime.fromISO(persistentCacheEntry
+      ? persistentCacheEntry.value
+      : defaultLastUpdateTime);
+    let windowEndTime = DateTime.utc();
+
+    // During backfilling, we process one day at a time to reduce roundtable runtime.
+    if (windowEndTime > lastUpdateTime.plus({ days: 1 })) {
+      windowEndTime = lastUpdateTime.plus({ days: 1 });
+    }
+
+    await AffiliateInfoTable.updateInfo(lastUpdateTime.toISO(), windowEndTime.toISO());
```

Comment:
Wrap these events in a transaction. There is a race condition:

1. current task reads persistent cache
2. another task overwrites persistent cache
3. current task updates affiliateInfoTable and persistent cache not knowing that the `lastUpdateTime` is stale

Reply:
Good point. To resolve this race condition, I could modify the query in AffiliateInfoTable.updateInfo to:
1. get cache time and lock row for update
2. get block time
3. compare block height to cache time
4. current logic for updating and setting new cache time

Reply:
just using a transaction is sufficient. Getting block time doesnt even need to be included in the race conditions